webapp/javascript/components/Heatmap/utils.ts

```diff
@@ -89,42 +89,42 @@ export const getSelectionData = (
   };
 };
 
-export const getFormatter = (
-  format: 'value' | 'time',
-  min?: number,
-  max?: number
-) => {
-  switch (format) {
-    case 'value':
-      return (v: number) =>
-        v > 1000 ? `${(v / 1000).toFixed(1)}k` : v.toFixed(0);
-    case 'time':
-      return (v: number) => {
-        const date = new Date(v / 1000000);
-        // nanoseconds -> hours
-        const hours =
-          ((max as number) - (min as number)) / 60 / 60 / 1000 / 1000 / 1000;
-
-        return getTimelineFormatDate(date, hours);
-      };
-    default:
-      return () => '';
-  }
-};
+export const timeFormatter =
+  (min: number, max: number, timezone: string) => (v: number) => {
+    const d = getUTCdate(
+      new Date(v / 1000000),
+      timezone === 'utc' ? 0 : new Date().getTimezoneOffset()
+    );
+    // nanoseconds -> hours
+    const hours = (max - min) / 60 / 60 / 1000 / 1000 / 1000;
+
+    return getTimelineFormatDate(d, hours);
+  };
```

Comment:
I not a big fan of having this logic repeat in multiple places (https://github.com/pyroscope-io/pyroscope/issues/1521), can we extract common logic to Can we move that here https://github.com/pyroscope-io/pyroscope/blob/main/webapp/javascript/util/formatDate.ts
And here only have the part that's >strictly< related to Heatmap?

Reply:
ye, i will refactor it